project/apps/web/components/collab/CollaborativeEditor.tsx

```diff
@@ -183,13 +177,14 @@ const CollaborativeEditor = forwardRef<
       return () => {
         provider.disconnect();
         yRuntime.unobserve(handleYRuntimeChange);
+        provider.awareness?.off('change', handleAwarenessChange);
       };
     } else {
       setCollabLoading(false);
     }
   };
 
-  const endSession = () => {
+  const handleEndSession = () => {
     if (providerRef.current) {
       providerRef.current.setAwarenessField('sessionEnded', {
         endedBy: user?.id,
```

Comment:
is it possible to disconnect the hocuspocus provider when the session has ended?
so that whatever the user types wont affect the stored Yjs document in the db